EKF/vel_pos_fusion.cpp

```diff
@@ -102,6 +102,20 @@ void Ekf::fuseVelPosHeight()
 			// innovation gate size
 			gate_size[5] = fmaxf(_params.baro_innov_gate, 1.0f);
 
+			// Compensate for positive static pressure transients (negative vertical position innovations)
+			// casued by rotor wash ground interaction by applying a temporary deadzone to baro innovations.
+			float deadzone_start = 0.25f * _params.baro_noise;
+			float deadzone_end = deadzone_start + _params.gnd_effect_deadzone;
+			if (_control_status.flags.gnd_effect) {
+				if (_vel_pos_innov[5] < -deadzone_start) {
+					if (_vel_pos_innov[5] <= -deadzone_end) {
+						_vel_pos_innov[5] += deadzone_end;
```

Comment:
This is more of a question, but can you explain the logic here? Why is it shifted by deadzone_end rather than just constrained within the dead zone?

Reply:
I don't understand the question.

Reply:
When _vel_pos_innov[5] (z pos innovation) is below the deadzone end you increase it by deadzone_end. Why bump the innovation by deadzone_end instead of just limiting it to deadzone_end?

Reply:
Because it prevents a possible runaway of the inertial height in that direction and still allows for temporary baro height 'spikes' up to the value set by the parameter.